### PR TITLE
Backup Plugin: add a dependency on the automattic/jetpack-composer-plugin package

### PR DESCRIPTION
**Pull request by kbrown9**
#### Changes proposed in this Pull Request:
* Add a dependency on the automattic/jetpack-composer-plugin package. This package will install `jetpack-library` packages in the `jetpack_vendor` directory, which will allow the packages' strings to be extracted for translation.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Checkout this branch.
2. Delete the existing vendor folder (`projects/plugins/backup/vendor`).
3. In the `projects/plugins/backup` directory, run `composer update automattic/jetpack-composer-plugin`.
4. Verify that 21 packages have been installed in the `projects/plugins/backup/jetpack_vendor/automattic` directory.
5. Verify that 3 packages (`jetpack-autoloader`, `jetpack_changelogger`, and `jetpack-composer-plugin`) have been installed in the `projects/plugins/backup/vendor/automattic` directory.

##### Check the Production build
1. Download the production build of this branch from the Jetpack beta build server: https://betadownload.jetpack.me/data/jetpack-backup/update_backup_composer_package_i18n/jetpack-backup-dev.zip
2. Verify that 21 packages have been installed in the `jetpack_vendor/automattic` directory.
3. Verify that no packages have been installed in the `vendor/automattic` directory.

### CODE REVIEW

**Review by anomiex — DISMISSED** [review]
LGTM. Confirmed that the built artifact doesn't have any unexpected changes.

**Comment by kbrown9** [comment]
This needs another review because I had to update the `composer.lock` file. I've enabled auto-merge so this will merge as soon as it has an approval.

**Review by anomiex — APPROVED** [review]

**Pull request merged**
